src/directories.ts

```diff
@@ -0,0 +1,58 @@
+import { ChildProcess, spawn } from 'child_process'
+import { workspace } from 'coc.nvim'
+import FilesList, { Task as FilesTask } from './files'
+import { executable } from './util'
+
+class Task extends FilesTask {
+  protected post_process_hook(process: ChildProcess) {
+    let post_process = spawn('sh', ['-c', 'xargs -I {} gdirname {} | sort | uniq'])
```

Comment:
You can use `sort --unique` instead of `sort | uniq`. But isn't there a better way than spawning another shell?

Reply:
Thanks @kidonng. I've noticed it's very slow on larger projects. Hopefully I'll have time to improve it soon.